src/QMCWaveFunctions/LCAO/LCAOrbitalBuilder.cpp

```diff
@@ -572,6 +572,7 @@ std::unique_ptr<SPOSet> LCAOrbitalBuilder::createSPOSetFromXML(xmlNodePtr cur)
   }
 #endif
 
+  sposet->finalizeConstruction();
```

Comment:
the intention here was to make sure the coefs are correct on the device when the LCAOrbitalSet is created (rather than putting C->updateTo() before each AO->MO gemm), but maybe this is not the right way to do it.  If I understand it correctly, all of the LCAOrbitalSets in the RefVectorWithLeader<SPOSet> that we use in the mw_ functions have a shared pointer to the same coefs, so it is probably better to just do the H2D update once from the leader whenever that RefVectorWithLeader<SPOSet> is created.

This is also where [some tests](https://github.com/QMCPACK/qmcpack/actions/runs/7544922523/job/20539381951?pr=4878#step:6:2032) are failing, but I haven't tracked down exactly what's going wrong there yet.

Reply:
Fixed